### PR TITLE
make_ule: Support enums with min and max

### PR DESCRIPTION
**Pull request by sffc**
I made an enum where I wanted this in #5392

Shouldn't make an impact on existing enums because `if unsized_int < 0` is always false so it should compile out.

If you would rather not land this, I can re-work #5392 to make an enum starting at zero

### CODE REVIEW

**Review by Manishearth — APPROVED** [review]
Nice

File: utils/zerovec/derive/src/make_ule.rs [hunk 1]
```diff
@@ -108,11 +115,31 @@ fn make_ule_enum_impl(
...
+                            variant.span(),
+                            "#[make_ule] only supports discriminants from 0 to 255",
+                        )
+                        .to_compile_error();
+                    }
+                };
+                match min {
+                    Some(x) if x < n => {}
+                    _ => {
+                        min = Some(n);
```
[the remark this answers is not in the document]
Comment (sffc) [reply utils/zerovec/derive/src/make_ule.rs:131]: Calculating `max` updates the `not_found` list for all values starting from 0, so I don't think we need to do anything special when `min` is updated.

File: utils/zerovec/derive/Cargo.toml [hunk 2]
```diff
@@ -32,3 +32,18 @@ serde = { workspace = true, features = ["derive"] }
...
+
+[[example]]
+name = "make"
+harness = false
+test = true
+
+[[example]]
+name = "make_var"
+harness = false
+test = true
```
Comment (sffc) [inline utils/zerovec/derive/Cargo.toml:49]: Seems like we need this only in `zerovec_derive`. It seems the examples in most other crates are already running in CI. Not clear what makes this crate special: maybe that it is a proc macro? And this is the only proc macro crate with examples that have assertions.

Comment (Manishearth) [reply utils/zerovec/derive/Cargo.toml:49]: yeah probably

**Review by Manishearth — APPROVED** [review]

**Pull request merged**